src/EditorFeatures/Core/EditorConfigSettings/Updater/NamingStyles/EditorConfigNamingStylesExtensions.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles;
+using Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Data;
+using Microsoft.CodeAnalysis.EditorConfig.Parsing.NamingStyles;
+using Microsoft.CodeAnalysis.NamingStyles;
+
+namespace Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Updater
+{
+    internal static class EditorConfigNamingStylesExtensions
+    {
+        public static bool TryGetParseResultForRule(
+            this EditorConfigNamingStyles editorConfigNamingStyles,
+            NamingStyleSetting namingStyleSetting,
+            [NotNullWhen(true)] out NamingStyleOption? namingStyleOption)
+        {
+            namingStyleOption = null;
+            foreach (var (option, optionAsNamingStyle) in editorConfigNamingStyles.Rules.AsNamingStyleSettings())
+            {
+                if (AreSameRule(optionAsNamingStyle, namingStyleSetting))
+                {
+                    namingStyleOption = option;
+                    return true;
+                }
+            }
+
+            return false;
+
+            static bool AreSameRule(NamingStyleSetting left, NamingStyleSetting right)
```

Comment:
We need to compare them ignoring their names because they can change between VS and editorconfig. VS allows spaces in the names whereas editorconfig does not